cmd/cmd_util.go

```diff
@@ -0,0 +1,158 @@
+		rbdImageName := kubernetes.GetFlexVolumeName(pv)
+		if rbdImageName == "" {
+			// TODO continue in case rbdImageName is empty.
+			logger.ErrorLog("rbdImageName cannot be empty in Flex PV object: %v", pv)
+		}
+		logger.DebugLog("flex rbdImageName name %v ", rbdImageName)
+
+		// Delete the PVC object
+		err = kubernetes.DeletePVC(client, &p) // nolint:gosec // skip gosec as pvc is accessed via it's reference.
+		if err != nil {
+			// TODO have an option to revert back if some Delete fails?
+			logger.ErrorLog("failed to Delete PVC object %s \n", p.Name)
+			continue
+		}
+		// Create new PVC with same name in destination storageclass
+		csiPVC := kubernetes.GenerateCSIPVC(destinationStorageClass, &p) // nolint:gosec // skip gosec as pvc is accessed via it's reference.
+		logger.DebugLog("structure of the generated PVC in destination Storageclass \n %v ", csiPVC)
+
+		// Create new CSI PVC
+		csiPV, err := kubernetes.CreatePVC(client, csiPVC, 5)
+		if err != nil {
+			logger.ErrorLog("failed to Create CSI PVC object %s : %v", p.Name, err)
+			continue
+		}
+		logger.DebugLog("new PVC with same name created in CSI. Persistent Volume: %v ", csiPV)
+
+		// Extract new volume name from CSI PV
+		csiRBDImageName := kubernetes.GetCSIVolumeName(csiPV)
+		if csiRBDImageName == "" {
+			logger.ErrorLog("csiRBDImageName cannot be empty in PV object")
+		}
+		logger.DebugLog("csi new volume name: %v ", csiRBDImageName)
+		poolName := kubernetes.GetCSIPoolName(csiPV)
+		if poolName == "" {
+			logger.ErrorLog("poolName cannot be empty in PV object")
+		}
+		logger.DebugLog("csi poolname: %v ", poolName)
+
+		// Create new connection
+		conn, err := createClusterConnection(client, csiPV)
+		if err != nil {
+			logger.ErrorLog("failed to get cluster config %v \n", err)
+		}
+		defer func() {
+			err = conn.Destroy()
+			if err != nil {
+				logger.ErrorLog("failed to destroy the connection: %v", err)
+			}
+		}()
+
+		// Delete the CSI volume in ceph cluster
+		err = conn.RemoveVolumeAdmin(poolName, csiRBDImageName)
+		if err != nil {
+			logger.ErrorLog("err in RemoveVolume: %v", err)
+		}
+		logger.DebugLog("successfully removed volume %s", csiRBDImageName)
+
+		// Rename old ceph volume to new CSI volume
+		err = conn.RenameVolume(csiRBDImageName, rbdImageName)
+		if err != nil {
+			logger.ErrorLog("err in RenameVolume: %v", err)
+		}
+		logger.DebugLog("successfully renamed volume %s -> %s", csiRBDImageName, rbdImageName)
+
+		// Delete old PV object
+		err = kubernetes.DeletePV(client, pv)
+		if err != nil {
+			logger.ErrorLog("err in DeletePV: %v", err)
+		}
+		logger.DebugLog("deleted persistent volume %s", pv)
+	}
+
+	return err
+}
+
+// creareClusterConnection creates a connection to the ceph cluster.
+func createClusterConnection(client *k8s.Clientset, csiPV *v1.PersistentVolume) (*rbd.Connection, error) {
+	poolName := kubernetes.GetCSIPoolName(csiPV)
+	if poolName == "" {
+		logger.ErrorLog("poolName cannot be empty in PV object")
```

Comment:
Why not an error instead of logging?

Reply:
Thanks for pointing it out! Updated code to return error in case of failure with appropriate error messages; Also log errors when required.